### PR TITLE
Adds a varset_callback to reduce the proc amount

### PR DESCRIPTION
**Pull request by DominikPanic**
Quite a few callbacks are just setting one variable to true/false, this will make further unshitcoding and porting useful.

### CODE REVIEW

**Comment by psykzz** [comment]
Do you have an example of how to use `VARSET_LIST_CALLBACK`?

**Comment by DominikPanic** [comment]
> 
> 
> Do you have an example of how to use `VARSET_LIST_CALLBACK`?

No, /tg/ doesn't use it either as far as I can tell, but it's useful to have regardless.

File: code/__DEFINES/callbacks.dm [hunk 1]
```diff
@@ -2,3 +2,6 @@
 
 #define CALLBACK new /datum/callback
 #define INVOKE_ASYNC world.ImmediateInvokeAsync
+
+#define VARSET_LIST_CALLBACK(target, var_name, var_value) CALLBACK(GLOBAL_PROC, /proc/___callbackvarset, ##target, ##var_name, ##var_value)
+#define VARSET_CALLBACK(datum, var, var_value) CALLBACK(GLOBAL_PROC, /proc/___callbackvarset, ##datum, NAMEOF(##datum, ##var), ##var_value)
```
Comment (Rohesie) [inline code/__DEFINES/callbacks.dm]:
```suggestion
#define VARSET_CALLBACK(datum, var, var_value) (CALLBACK(GLOBAL_PROC, /proc/___callbackvarset, (##datum), (NAMEOF(##datum), (##var)), (##var_value)))
```
Make use and abuse of parenthesis on macros.

**Comment by Rohesie** [comment]
This is very, very nice.

File: code/__DEFINES/callbacks.dm [hunk 2]
```diff
@@ -2,3 +2,6 @@
 
 #define CALLBACK new /datum/callback
 #define INVOKE_ASYNC world.ImmediateInvokeAsync
+
+#define VARSET_LIST_CALLBACK(target, var_name, var_value) CALLBACK(GLOBAL_PROC, /proc/___callbackvarset, ##target, ##var_name, ##var_value)
```
Comment (Rohesie) [inline code/__DEFINES/callbacks.dm]:
```suggestion
#define VARSET_LIST_CALLBACK(target, var_name, var_value) (CALLBACK(GLOBAL_PROC, /proc/___callbackvarset, (##target), (##var_name), (##var_value)))
```

**Pull request merged**